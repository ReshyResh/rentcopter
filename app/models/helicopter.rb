class Helicopter < ApplicationRecord
  mount_uploader :avatar, AvatarUploader
  belongs_to :user
  has_many :reservations, dependent: :destroy

  validates :name, :description, :model, :image, presence: true
end
