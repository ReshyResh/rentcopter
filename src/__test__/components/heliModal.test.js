import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import HeliModal from '../../components/HeliModal';
import helicopter from '../mockHeli';
import store from '../../redux/configureStore';

it('Helicopter modal component renders correctly', () => {
  const tree = renderer
    .create(<Router>
      <Provider store={store}>
        <HeliModal helicopter={helicopter} />
      </Provider>
            </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
