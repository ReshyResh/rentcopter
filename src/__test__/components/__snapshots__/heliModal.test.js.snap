// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helicopter modal component renders correctly 1`] = `
<div>
  <div
    className="w-full h-32 sm:w-2/3 md:w-full mx-auto md:h-60 rounded-2xl flex hover:shadow-2xl hover:shadow-gray-600 transition-all"
    style={
      Object {
        "backgroundImage": "url(https://luxcior.com/wp-content/uploads/2021/03/Beverly-Hills-Magazine-luxury-helicopters-for-sale-private-helicopter-EUROCOPTERS-5.jpg)",
        "backgroundPosition": "center center",
        "backgroundSize": "cover",
        "borderRadius": 10,
        "display": "flex",
        "flexFlow": "column",
        "justifyContent": "flex-start",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root h-3/4 border-b border-gray-600 css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "alignItems": "start",
          "display": "flex",
          "flexFlow": "column",
          "justifyContent": "flex-start",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        className="h-10 w-full bg-gray-900 bg-opacity-75 rounded-t-md"
      >
        <h1
          className="cool-title ml-4 my-auto text-left h-full"
        >
          Copter-X
        </h1>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className=" bg-gray-900 w-full h-1/4 flex flex-row-reverse justify-between bg-opacity-50"
    >
      <div
        className="h-full flex justify-center"
      >
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root 
 !h-1/2
 !sm:h-1/2 
 !my-auto 
 !mr-3
 !rounded-full 
 !text-sm
 !bg-opacity-75
 !w-24
 !font-bold 
 !bg-yellow-600
 !hover:bg-indigo-700
 !transition-all
 !text-white css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          RESERVE
        </button>
      </div>
      <div
        className="ml-3 my-auto flex"
      >
        <svg
          fill="rgb(0,200,50)"
          height="35"
          viewBox="0 0 24 24"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 10.999c-2.209 0-4 1.792-4 4.001s1.791 4 4 4 4-1.791 4-4-1.791-4.001-4-4.001zm0 7.001c-1.654 0-3-1.346-3-3s1.346-3.001 3-3.001 3 1.347 3 3.001-1.346 3-3 3zm.167-1.351v.351h-.334v-.333c-.344-.006-.702-.088-1-.242l.152-.548c.254.099.574.202.861.202l.213-.022c.383-.086.462-.48.039-.67-.311-.145-1.26-.269-1.26-1.081 0-.455.346-.861.994-.95v-.356h.334v.339c.24.006.512.049.814.141l-.121.548c-.232-.081-.487-.156-.738-.156l-.076.002c-.496.029-.541.459-.193.639.569.268 1.314.467 1.314 1.181.001.573-.446.878-.999.955zm-13.167-5.649c0 1.656 1.344 3 3 3s3-1.344 3-3-1.344-3-3-3-3 1.344-3 3zm3.883-1.368l-.108.493c-.23-.08-.485-.154-.733-.139-.446.026-.486.413-.174.575.514.242 1.182.42 1.182 1.063.002.515-.401.791-.898.86v.315h-.302v-.299c-.311-.005-.632-.079-.898-.217l.135-.493c.287.11.669.229.968.162.345-.078.415-.433.034-.604-.279-.129-1.133-.242-1.133-.973 0-.409.312-.775.895-.855v-.319h.301v.305c.215.005.459.043.731.126zm4.319 7.368h-15.202v-12h20v4.799c-.709 0-1.384.145-2 .402v-1.219c-.959-.42-1.395-1.023-1.814-1.982h-12.372c-.419.959-.855 1.562-1.814 1.982v4.036c.959.42 1.395 1.022 1.814 1.982h10.986c0 .708.144 1.384.402 2z"
          />
        </svg>
        <p
          className="ml-2 text-xs text-white "
          style={
            Object {
              "fontFamily": "Montserrat",
            }
          }
        >
          Price per day:
          <br />
          <span
            className="text-md font-bold"
          >
            500
            $
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
